### PR TITLE
Fix an expired todo test

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
I put a specific error there, if someone changes getFileInfo, they can update the Error info

File: tests_integration/__tests__/file-info.js [hunk 1]
```diff
@@ -106,16 +106,17 @@ describe("extracts file-info with inferredParser=foo when a plugin is hand-picke
...
 test("API getFileInfo with no args", () => {
-  // TODO: change this to `rejects.toThrow()` when we upgrade to Jest >= 22
-  // https://github.com/facebook/jest/issues/3601
-  expect.assertions(1);
-  return prettier.getFileInfo().catch(err => {
-    expect(err).toBeDefined();
-  });
+  expect(prettier.getFileInfo()).rejects.toThrow(
+    // this error info might change depends on how getFileInfo works
+    new TypeError(`Cannot read property 'replace' of undefined`)
```
Comment (lydell) [inline tests_integration/__tests__/file-info.js]: Where does this error message come from?

Comment (fisker) [reply tests_integration/__tests__/file-info.js]: see comment above.

for now its come from undefined.replace

too strict?

Comment (fisker) [reply tests_integration/__tests__/file-info.js]: or maybe we should throw a error with specific info?

File: tests_integration/__tests__/file-info.js [hunk 2]
```diff
@@ -106,16 +106,15 @@ describe("extracts file-info with inferredParser=foo when a plugin is hand-picke
 });
 
 test("API getFileInfo with no args", () => {
-  // TODO: change this to `rejects.toThrow()` when we upgrade to Jest >= 22
-  // https://github.com/facebook/jest/issues/3601
-  expect.assertions(1);
-  return prettier.getFileInfo().catch(err => {
-    expect(err).toBeDefined();
-  });
+  expect(prettier.getFileInfo()).rejects.toThrow(
```
Comment (SimenB) [inline tests_integration/__tests__/file-info.js]: you need to `return` or `await` it - it's async

Comment (fisker) [reply tests_integration/__tests__/file-info.js]: you are right, I forgot, fixed

**Comment by fisker** [comment]
update to use return , it's more readable https://github.com/prettier/prettier/issues/6668

**Comment by fisker** [comment]
update to use return , it's more readable https://github.com/prettier/prettier/issues/6668

File: src/common/get-file-info.js [hunk 3]
```diff
@@ -19,6 +19,14 @@ const path = require("path");
  * internally by the method wrapper. See withPlugins() in index.js.
  */
 function getFileInfo(filePath, opts) {
+  if (typeof filePath !== "string") {
+    return Promise.reject(
```
Comment (evilebottnawi) [inline src/common/get-file-info.js]: Just use `throw new TypeError`, no need use `Promise.reject`

Comment (fisker) [reply src/common/get-file-info.js]: no, that would  be a breaking  change since thi function I not async

Comment (evilebottnawi) [reply src/common/get-file-info.js]: my mistake (i think it is async :smile: )

**Pull request merged**